### ISSUE TITLE
Build Tool Request: Snowpack

### ISSUE THREAD

**Issue by FredKSchott** [jakoblind/webpack-autoconf#118]
First off, thanks for the great project! 

[From snowpack.dev:](https://snowpack.dev/)

> Snowpack is a modern frontend build tool for faster web development. It replaces heavier, complex bundlers like webpack or Parcel in your development workflow.
> 
> Snowpack leverages JavaScript’s native module system (known as ESM) to create a first-of-its-kind build system that never builds the same file twice. Snowpack pushes changes instantly to the browser, saving you hours of development time traditionally spent waiting around for your bundler.

We've been seeing a ton of adoption recently, especially in the "beginner" / "getting started easily" space that this site seems to focus on. Dr. Axel Rauschmayer recently used Snowpack in his "minimal react" tutorial, for example: https://2ality.com/2020/08/minimal-react.html

Is there any interest in adding Snowpack to the site? I'm happy to do the work (or at least explore it, to start) but would want your blessing before moving ahead on a PR.

**Comment by jakoblind** [comment]
I've never heard of snowpack, but it seems like a solid project! A PR with snowpack support is very welcomed. 

My advice is to send multiple smaller PRs, or at least share your work along the way if it gets big. we can discuss in this thread.

**Comment by jakoblind** [comment]
Hi @FredKSchott. I've looked into snowpack and it looks really promising, thanks for letting me know about it! I have implemented support for React and some of the features that were easy to move over from webpack. I wanted to share this with you before I merge it into master. You can run it [live here](https://5f96bfe56f50d20007313d77--naughty-ramanujan-bf38b8.netlify.app/snowpack), and the code in is in #121 . 

Next, I'm going to implement svelte, and most of what's in the [recipes section in snowpack docs](https://www.snowpack.dev/#recipes) 

Please let me know if you think there are any other features I should prioritize.

**Comment by jakoblind** [comment]
Alright, I just shipped the first version of snowpack! With React and some other features.

**Comment by FredKSchott** [comment]
Ah exciting!!! That's great to see! Will play around a bit today and report back with a PR if I find any changes/fixes.

**Comment by jakoblind** [comment]
Awesome! I wanted to ask your opinion on something. I would assume that most people would want to add `@snowpack/plugin-webpack` to bundle their projects, so I'm tempted to add that as a default for all Snowpack projects. The option is to instead add it as a feature that the user must to select.

Also, I think most users want `@snowpack/plugin-optimize` to minify their build. I think I want to add it by default also. 

And then I'm thinking of adding documentation (the question mark icon) describing what's going on. 

If you have any thoughts on this I'm happy to hear about it :)

**Comment by FredKSchott** [comment]
Yes, I think @snowpack/plugin-webpack is a good default, and that also minifies your code so that you don't need to use it with `@snowpack/plugin-optimize` (just one or the other). 

Happy to help write or edit the question mark helper text!

**Comment by jakoblind** [comment]
Good! Then I'll make that a default.

I just realized I don't have any general build tool help button, will need to put on the UX hat and think where to put it. Thanks for offering to help with that text, will come back to you when I have a first draft ready :)

**Comment by Wanaco** [comment]
Hi @jakoblind,

When selecting snowpack + react + tailwindcss I have notice that the "@snowpack/plugin-build-script" is referenced in the _snowpack.config.json_ file but it is missing from the devDependencies.  It took me some time to figure out the error it was spitting.

I'm still a beginner in the whole web stack framework world.  Your tool is intuitive and has helped me demistify the project configurations.  Great Work!

**Comment by jakoblind** [comment]
Hi @Wanaco 

Thanks for letting me know about this bug, I just fixed it! Happy you like the project :)
